install/gcp/installer/util/gcloud.py

```diff
@@ -230,22 +238,32 @@ def grant_server_svc_acct_roles(enable_write,
         bool: Whether or not a role script has been generated.
     """
 
-    utils.print_banner('Assigning Roles To The GCP Service Account',
+    utils.print_banner('Assigning Roles To The GCP Service Account '
+                       'for the forseti project',
                        gcp_service_account)
-    access_target_roles = constants.GCP_READ_IAM_ROLES
-    if enable_write:
-        access_target_roles.extend(constants.GCP_WRITE_IAM_ROLES)
-
-    roles = {
-        '%ss' % access_target: access_target_roles,
-        'forseti_project': constants.PROJECT_IAM_ROLES_SERVER,
-        'service_accounts': constants.SVC_ACCT_ROLES,
-    }
+    roles = {}
+    roles['forseti_project'] = constants.PROJECT_IAM_ROLES_SERVER
+    roles['service_accounts'] = constants.SVC_ACCT_ROLES
+    target_id = project_id
 
     has_role_script_rest = _grant_svc_acct_roles(
         target_id, project_id, gcp_service_account,
         user_can_grant_roles, roles)
 
+    for resource in resources:
+        utils.print_banner('Assigning Roles To The GCP Service Account '
+                           'for the resource[' + resource + ']',
+                           gcp_service_account)
+        access_target, target_id = resource.split('/')
+        access_target_roles = constants.GCP_READ_IAM_ROLES
+        if enable_write:
+            access_target_roles.extend(constants.GCP_WRITE_IAM_ROLES)
+        roles = {}
+        roles[access_target] = access_target_roles
+        has_role_script_rest = _grant_svc_acct_roles(
```

Comment:
Does _grant_svc_acct_roles handle granting roles on folder/project level?

Reply:
I will re-test, but access_target will use gcloud projects add-iam-policy-binding vs gcloud alpha resource-manager folders add-iam-policy-binding

Reply:
Thanks, since we are using the composite resource id flag now, just wanted to make sure it also works with folder level permission :)

Reply:
I've confirmed that this works at both folder and project levels. The composite-root-resources passed in the args must be "folders/FOLDER_ID" NOT FOLDER_NAME

We could add a check to confirm it is an ID vs NAME, but this type of check is not a common pattern in the code today. I do feel it would add value, but may be best left for another bug?

Here is an excerpt from folder based install 
(Note: I've truncated and masked the real svc acct, project, and folder id)

+-------------------------------------------------------------------------------
|  Assigning Roles To The GCP Service Account for the resource[folders/*580]
|  forseti-server-gcp-*@*-forseti-01.iam.gserviceaccount.com
+-------------------------------------------------------------------------------

Assigning roles/appengine.appViewer on *580... assigned
Assigning roles/bigquery.metadataViewer on *580... assigned
Assigning roles/browser on *580... assigned
Assigning roles/cloudasset.viewer on *580... assigned
Assigning roles/cloudsql.viewer on *580... assigned
Assigning roles/compute.networkViewer on *580... assigned
Assigning roles/iam.securityReviewer on *580... assigned
Assigning roles/orgpolicy.policyViewer on *580... assigned
Assigning roles/servicemanagement.quotaViewer on *580... assigned
Assigning roles/serviceusage.serviceUsageConsumer on *580... assigned
Assigning roles/compute.securityAdmin on *580... assigned